core/src/main/java/org/apache/accumulo/core/lock/ServiceLockPaths.java

```diff
@@ -362,15 +377,16 @@ private Set<ServiceLockPath> get(final String serverType, Optional<String> resou
             final ZcStat stat = new ZcStat();
             final ServiceLockPath slp =
                 parse(Optional.of(serverType), typePath + "/" + group + "/" + server);
-            if (slp.getType().equals(Constants.ZDEADTSERVERS)
-                && (address.isEmpty() || address.orElseThrow().toString().equals(server))) {
-              // Dead TServers don't have lock data
-              results.add(slp);
-            } else {
-              Optional<ServiceLockData> sld = ServiceLock.getLockData(cache, slp, stat);
-              if (!sld.isEmpty()
-                  && (address.isEmpty() || address.orElseThrow().toString().equals(server))) {
+            if (address.isEmpty() || address.orElseThrow().toString().equals(server)) {
+              if (!withLock || slp.getType().equals(Constants.ZDEADTSERVERS)) {
+                // Dead TServers don't have lock data
                 results.add(slp);
+              } else {
+                Optional<ServiceLockData> sld = ServiceLock.getLockData(cache, slp, stat);
+                if (!sld.isEmpty()
+                    && (address.isEmpty() || address.orElseThrow().toString().equals(server))) {
```

Comment:
Seems like the address already being checked by the outer if stmt and could be removed here.

```suggestion
                if (!sld.isEmpty()) {
```

Reply:
Good catch. Removed the redundant checks in 75252c7